spark/src/test/scala/org/apache/spark/sql/delta/typewidening/TypeWideningTableFeatureSuite.scala

```diff
@@ -454,6 +457,43 @@ trait TypeWideningTableFeatureTests extends RowTrackingTestUtils with TypeWideni
     )
   }
 
+  testSparkLatestOnly(
+    "helpful error when reading type changes not supported yet during preview") {
+    sql(s"CREATE TABLE delta.`$tempDir` (a int) USING DELTA")
+    val metadata = new MetadataBuilder()
+      .putMetadataArray("delta.typeChanges", Array(
+        new MetadataBuilder()
+          .putString("toType", "long")
+          .putString("fromType", "int")
+          .putLong("tableVersion", 1)
+          .build()
+      )).build()
+
+    // Delta 3.2/3.3 doesn't support changing type from int->long, we manually commit that type
+    // change to simulate what Delta 4.0 could do.
+    deltaLog.withNewTransaction { txn =>
+      txn.commit(
```

Comment:
Commit a new protocol with `TypeWideningPreviewTableFeature` supported?

Reply:
Type widening is enabled automatically by default on all tables created in this test suite: https://github.com/delta-io/delta/blob/b1e4a033ba81c57ba64a3705ded73d6a93d44497/spark/src/test/scala/org/apache/spark/sql/delta/typewidening/TypeWideningTestMixin.scala#L43
This adds the preview table feature to the table

Is that what you were pointing out?

Reply:
Yup.